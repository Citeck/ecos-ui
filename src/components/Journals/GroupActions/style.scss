@import '../../../styles/constants';

.ecos-group-actions {
  $root: &;

  &__opened-focus #{$root}__control_new {
    background-color: var(--primary-color) !important;
    border-radius: 6px;

    > i,
    > span {
      color: $color-white !important;
    }
  }

  &__dropdown {
    margin-right: 15px;
    max-width: 300px;
    transition: width;

    &-menu &-item {
      &_disabled {
        cursor: default;
        pointer-events: none;
        color: $color-gray-5;
      }
    }

    &-item {
      &:not(&_query) + &_query {
        border-top: $color-gray-2 2px solid;
      }
    }

    &_new .ecos-btn > i {
      font-size: 14px;
      color: $color-gray-5;
    }
    &_new:hover .ecos-btn > i {
      color: $color-blue-light;
    }
  }

  &__control.ecos-btn {
    overflow: hidden;
    width: 100%;
    text-overflow: ellipsis;
    font-size: 14px;
    padding: 0 15px;

    & .icon_semantic {
      font-size: 17px;
      color: $color-gray-5;
    }
  }

  &__control_new.ecos-btn {
    border-radius: 6px;
    overflow: hidden;
    width: 100%;
    text-overflow: ellipsis;
    font-size: 11px;
    padding: 0 12px;
    height: 28px;
    background-color: $color-gray-25;
    line-height: 18.6px;

    &:focus {
      background-color: $color-gray-25;

      > span {
        color: $text-color-gray;
      }
    }

    &:hover {
      background-color: var(--primary-color);

      > i,
      > span {
        color: $color-white;
      }
    }
  }

  &__control_mobile.ecos-btn {
    padding: 0 10px;
  }
}
