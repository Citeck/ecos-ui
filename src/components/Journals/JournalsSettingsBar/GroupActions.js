import React from 'react';
import { connect } from 'react-redux';
import get from 'lodash/get';
import isEmpty from 'lodash/isEmpty';

import { t } from '../../../helpers/export/util';
import { wrapArgs } from '../../../helpers/redux';
import { execRecordsAction, setSelectAllRecords, setSelectedRecords } from '../../../actions/journals';
import { selectGroupActionsProps } from '../../../selectors/journals';
import { DropdownOuter } from '../../common/form';

const Labels = {
  SELECTED: 'journal.group-actions.selected-records'
};

const mapStateToProps = (state, props) => {
  const ownState = selectGroupActionsProps(state, props.stateId);

  return {
    isMobile: get(state, 'view.isMobile'),
    ...ownState
  };
};

const mapDispatchToProps = (dispatch, props) => {
  const w = wrapArgs(props.stateId);

  return {
    setSelectedRecords: records => dispatch(setSelectedRecords(w(records))),
    setSelectAllRecords: need => dispatch(setSelectAllRecords(w(need))),
    execRecordsAction: (records, action, context) => dispatch(execRecordsAction(w({ records, action, context })))
  };
};

const GroupActions = props => {
  const {
    isMobile,
    selectAllRecordsVisible,
    selectAllRecords,
    grid,
    selectedRecords,
    onExecuteAction,
    className,

    onSelectAll,
    forwardedRef
  } = props;

  if (isEmpty(selectedRecords) && !selectAllRecords) {
    return null;
  }

  const total = get(grid, 'total', 0);
  const recordsActions = get(grid, 'actions.forRecords.actions', {});
  const queryActions = get(grid, 'actions.forQuery.actions', {});

  const onChange = () => {};

  const onSelectAllRecords = () => {
    const { setSelectAllRecords, selectAllRecords, setSelectedRecords } = this.props;

    setSelectAllRecords(!selectAllRecords);

    if (!selectAllRecords) {
      setSelectedRecords([]);
    }
  };

  const onExecuteGroupAction = action => {
    const { selectAllRecords } = this.props;

    if (!selectAllRecords) {
      const records = get(this.props, 'selectedRecords', []);

      this.props.execRecordsAction(records, action);
    } else {
      const query = get(this.props, 'grid.query');

      this.props.execRecordsAction(query, action);
    }
  };

  return (
    <>
      <DropdownOuter
        isStatic
        valueField={'id'}
        titleField={'pluralName'}
        keyFields={['id', 'formRef', 'pluralName']}
        source={recordsActions}
        controlLabel={t(Labels.SELECTED, { selected: selectAllRecordsVisible ? total : selectedRecords.length, total })}
        className="group-actions__dropdown"
        menuClassName="999"
        controlClassName="ecos-btn_hover_blue2 ecos-btn_grey3 group-actions__control"
        onChange={onChange}
      />

      <DropdownOuter
        isStatic
        valueField={'id'}
        titleField={'pluralName'}
        keyFields={['id', 'formRef', 'pluralName']}
        source={queryActions}
        controlLabel={'Результат фильтрации'}
        className="group-actions__dropdown"
        menuClassName=""
        controlClassName="ecos-btn_hover_blue2 ecos-btn_grey3 group-actions__control"
        onChange={onChange}
      />
    </>
  );
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(GroupActions);
