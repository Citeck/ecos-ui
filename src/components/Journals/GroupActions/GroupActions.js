import React, { useCallback, useEffect, useState } from 'react';
import classNames from 'classnames';
import { connect } from 'react-redux';
import get from 'lodash/get';
import isEmpty from 'lodash/isEmpty';
import isEqual from 'lodash/isEqual';
import uniqueId from 'lodash/uniqueId';
import isNil from 'lodash/isNil';

import { t } from '../../../helpers/export/util';
import { wrapArgs } from '../../../helpers/redux';
import { execRecordsAction } from '../../../actions/journals';
import { selectGroupActionsProps } from '../../../selectors/journals';
import { DropdownOuter } from '../../common/form';
import { IcoBtn, TwoIcoBtn } from '../../common/btns';
import { Tooltip } from '../../common';

import './style.scss';

const Labels = {
  SELECTED: 'journal.group-actions.selected-records.label',
  SELECTED_SHORT: 'journal.group-actions.selected-records.label_short',
  SELECTED_COUNT: 'journal.group-actions.selected-records.label-count',
  RESULT: 'journal.group-actions.filter-result'
};

const TYPE_ACT = {
  QUERY: 'forQuery',
  RECORDS: 'forRecords'
};

const GroupActions = React.memo(
  props => {
    const [isOpenRecActions, setOpenRecActions] = useState(false);
    const [isOpenQueryActions, setOpenQueryActions] = useState(false);
    const [targetPrefix] = useState(uniqueId('group-actions-'));
    const [recordsActions, setRecordsActions] = useState([]);
    const [queryActions, setQueryActions] = useState([]);

    const { isMobile, grid, selectAllRecordsVisible, selectedRecords, execRecordsAction, isSeparateActionListForQuery } = props;

    const total = get(grid, 'total', 0);
    const selectedLen = selectedRecords.length;
    const selected = selectAllRecordsVisible ? total : selectedLen;
    const labelRecActionsCount = t(Labels.SELECTED_COUNT, { selected, total });
    const labelRecActions = t(isMobile ? Labels.SELECTED_SHORT : Labels.SELECTED, { data: labelRecActionsCount });

    useEffect(() => {
      const recordsActions = get(grid, 'actions.forRecords.actions', []).map(item => ({ ...item, _typeAct: TYPE_ACT.RECORDS }));
      const queryActions = get(grid, 'actions.forQuery.actions', []).map(item => ({ ...item, _typeAct: TYPE_ACT.QUERY }));

      if (!isSeparateActionListForQuery) {
        recordsActions.push(...queryActions);
      }

      setRecordsActions(recordsActions);
      setQueryActions(queryActions);
    }, [grid, isSeparateActionListForQuery]);

    const handleExecuteAction = useCallback(
      action => {
        if (action._typeAct === TYPE_ACT.QUERY) {
          execRecordsAction(grid.query, action);
        } else if (action._typeAct === TYPE_ACT.RECORDS) {
          execRecordsAction(selectedRecords, action);
        }
      },
      [grid, selectedRecords]
    );

    const getItemClassName = useCallback(
      action => {
        const disabled = action._typeAct === TYPE_ACT.RECORDS && (selectAllRecordsVisible || !selectedLen);
        return classNames('ecos-group-actions__dropdown-item', {
          'ecos-group-actions__dropdown-item_disabled': disabled,
          'ecos-group-actions__dropdown-item_query': action._typeAct === TYPE_ACT.QUERY
        });
      },
      [selectAllRecordsVisible, !!selectedLen]
    );

    const iconOpener = useCallback(flag => classNames('ecos-btn__i_right', { 'icon-small-up': flag, 'icon-small-down': !flag }), []);

    return (
      <>
        {!isEmpty(recordsActions) && (
          <DropdownOuter
            isStatic
            valueField={'id'}
            titleField={'pluralName'}
            keyFields={['id', 'formRef', 'pluralName', '_typeAct']}
            source={recordsActions}
            className="ecos-group-actions__dropdown"
            menuClassName="ecos-group-actions__dropdown-menu"
            getStateOpen={setOpenRecActions}
            onChange={handleExecuteAction}
            itemClassName={getItemClassName}
          >
            <Tooltip
              uncontrolled
              showAsNeeded
              target={targetPrefix + '-rec'}
              text={labelRecActions}
              contentComponent={labelRecActionsCount}
            >
              <IcoBtn
                invert
                className="ecos-btn_hover_blue2 ecos-btn_grey3 ecos-group-actions__control"
                icon={iconOpener(isOpenRecActions)}
                id={targetPrefix + '-rec'}
              >
                {labelRecActions}
              </IcoBtn>
            </Tooltip>
          </DropdownOuter>
        )}
        {isSeparateActionListForQuery && !isEmpty(queryActions) && (
          <DropdownOuter
            isStatic
            valueField={'id'}
            titleField={'pluralName'}
            keyFields={['pluralName', '_typeAct']}
            source={queryActions}
            className="ecos-group-actions__dropdown"
            controlClassName="ecos-btn_hover_blue2 ecos-btn_grey3 ecos-group-actions__control"
            menuClassName="ecos-group-actions__dropdown-menu"
            onChange={handleExecuteAction}
            getStateOpen={setOpenQueryActions}
          >
            <TwoIcoBtn
              invert
              className={classNames('ecos-btn_hover_blue2 ecos-btn_grey3 ecos-group-actions__control', {
                'ecos-group-actions__control_mobile': isMobile
              })}
              icons={[classNames({ 'icon-filter icon_semantic': isMobile }), iconOpener(isOpenQueryActions)]}
              id={targetPrefix}
            >
              {!isMobile && t(Labels.RESULT)}
            </TwoIcoBtn>
          </DropdownOuter>
        )}
      </>
    );
  },
  (prevProps, nextProps) =>
    isEqual(prevProps.stateId, nextProps.stateId) &&
    isEqual(prevProps.grid, nextProps.grid) &&
    isEqual(prevProps.selectAllRecordsVisible, nextProps.selectAllRecordsVisible) &&
    isEqual(prevProps.selectedRecords, nextProps.selectedRecords)
);

const mapStateToProps = (state, props) => {
  const ownState = selectGroupActionsProps(state, props.stateId);

  return {
    isMobile: isNil(props.isMobile) ? get(state, 'view.isMobile') : props.isMobile,
    isSeparateActionListForQuery: get(state, 'app.isSeparateActionListForQuery', false),
    ...ownState
  };
};

const mapDispatchToProps = (dispatch, props) => {
  const w = wrapArgs(props.stateId);

  return {
    execRecordsAction: (records, action, context) => dispatch(execRecordsAction(w({ records, action, context })))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(GroupActions);
