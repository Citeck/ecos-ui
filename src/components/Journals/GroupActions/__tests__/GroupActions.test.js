import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-canvas-mock';

import { GroupActions } from '../GroupActions';

configure({ adapter: new Adapter() });

describe('GroupActions tests', () => {
  const baseProps = {
    isMobile: false,
    isSeparateActionListForQuery: false,
    grid: {
      actions: {
        forRecords: {
          actions: []
        }
      }
    },
    columnsSetup: {},
    selectedRecords: [],
    excludedRecords: [],
    selectAllPageRecords: false,
    selectAllRecordsVisible: false,
    source: []
  };

  describe('<GroupActions />', () => {
    it('should render GroupActions component', () => {
      const component = shallow(<GroupActions {...baseProps} />);

      expect(component).toMatchSnapshot();
    });
  });

  describe('<GroupActions />', () => {
    it('should group action button is disabled', () => {
      const attachTo = document.createElement('div');

      document.body.appendChild(attachTo);

      const component = mount(<GroupActions {...baseProps} />, { attachTo });

      return Promise.resolve(component)
        .then(() => {
          const disabledComponents = component.find('.ecos-dropdown-outer_disabled');

          expect(disabledComponents.length).toBe(1);

          const groupActionSelector = component.find('.ecos-btn.ecos-group-actions__control');

          expect(groupActionSelector.props().disabled).toBeTruthy();
          expect(groupActionSelector.is('[disabled]')).toBe(true);
        })
        .then(() => component.unmount());
    });
  });

  describe('<GroupActions />', () => {
    it('should group action button is enabled', () => {
      const props = {
        ...baseProps,
        grid: {
          actions: {
            forRecords: {
              actions: [
                {
                  name: 'Скачать как Zip',
                  pluralName: 'Скачать как Zip',
                  type: 'download-zip',
                  theme: '',
                  icon: 'icon-download',
                  config: {},
                  confirm: null,
                  result: null,
                  features: { execForQuery: false, execForRecord: false, execForRecords: true },
                  id: 'download-zip',
                  __act_ctx__: { recordMask: 1, context: { mode: 'journal', scope: 'gr-requests-view' } }
                }
              ]
            }
          }
        }
      };
      const attachTo = document.createElement('div');

      document.body.appendChild(attachTo);

      const component = mount(<GroupActions {...props} />, { attachTo });

      return Promise.resolve(component)
        .then(() => {
          const disabledComponent = component.find('.ecos-dropdown-outer_disabled');

          expect(disabledComponent.length).toBe(0);

          const groupActionSelector = component.find('.ecos-btn.ecos-group-actions__control');

          expect(groupActionSelector.instance().getAttribute('disabled')).toBeNull();
          expect(groupActionSelector.props().disabled).toBe(false);
        })
        .then(() => component.unmount());
    });
  });
});
