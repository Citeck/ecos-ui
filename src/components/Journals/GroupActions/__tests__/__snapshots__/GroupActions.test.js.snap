// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupActions tests <GroupActions /> should render GroupActions component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UNDEFINED
    columnsSetup={Object {}}
    excludedRecords={Array []}
    grid={
      Object {
        "actions": Object {
          "forRecords": Object {
            "actions": Array [],
          },
        },
      }
    }
    isMobile={false}
    isSeparateActionListForQuery={false}
    selectAllPageRecords={false}
    selectAllRecordsVisible={false}
    selectedRecords={Array []}
    source={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <DropdownOuter
          className="ecos-group-actions__dropdown"
          disabled={true}
          getStateOpen={[Function]}
          isStatic={true}
          itemClassName={[Function]}
          keyFields={
            Array [
              "id",
              "formRef",
              "pluralName",
              "_typeAct",
            ]
          }
          menuClassName="ecos-group-actions__dropdown-menu"
          modifiers={
            Object {
              "flip": Object {
                "behavior": Array [
                  "bottom",
                  "top",
                  "right",
                  "left",
                ],
              },
            }
          }
          onChange={[Function]}
          outClassName="ecos-dropdown-outer-tooltip"
          placement="bottom-start"
          source={Array []}
          titleField="pluralName"
          trigger="click"
          valueField="id"
        >
          <Tooltip
            autohide={false}
            boundariesElement="window"
            contentComponent="journal.group-actions.selected-records.label-count"
            delay={0}
            hideArrow={false}
            isOpen={false}
            minWidthByContent={false}
            placement="top"
            showAsNeeded={true}
            target="group-actions-1-rec"
            text=""
            uncontrolled={true}
          >
            <IcoBtn
              className="ecos-btn_hover_blue2 ecos-btn_grey3 ecos-group-actions__control"
              disabled={true}
              icon="ecos-btn__i_right icon-small-down"
              id="group-actions-1-rec"
              invert={true}
              loaderType="points"
              loading={false}
            >
              
            </IcoBtn>
          </Tooltip>
        </DropdownOuter>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Tooltip
            autohide={false}
            boundariesElement="window"
            contentComponent="journal.group-actions.selected-records.label-count"
            delay={0}
            hideArrow={false}
            isOpen={false}
            minWidthByContent={false}
            placement="top"
            showAsNeeded={true}
            target="group-actions-1-rec"
            text=""
            uncontrolled={true}
          >
            <IcoBtn
              className="ecos-btn_hover_blue2 ecos-btn_grey3 ecos-group-actions__control"
              disabled={true}
              icon="ecos-btn__i_right icon-small-down"
              id="group-actions-1-rec"
              invert={true}
              loaderType="points"
              loading={false}
            >
              
            </IcoBtn>
          </Tooltip>,
          "className": "ecos-group-actions__dropdown",
          "disabled": true,
          "getStateOpen": [Function],
          "isStatic": true,
          "itemClassName": [Function],
          "keyFields": Array [
            "id",
            "formRef",
            "pluralName",
            "_typeAct",
          ],
          "menuClassName": "ecos-group-actions__dropdown-menu",
          "modifiers": Object {
            "flip": Object {
              "behavior": Array [
                "bottom",
                "top",
                "right",
                "left",
              ],
            },
          },
          "onChange": [Function],
          "outClassName": "ecos-dropdown-outer-tooltip",
          "placement": "bottom-start",
          "source": Array [],
          "titleField": "pluralName",
          "trigger": "click",
          "valueField": "id",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autohide": false,
            "boundariesElement": "window",
            "children": <IcoBtn
              className="ecos-btn_hover_blue2 ecos-btn_grey3 ecos-group-actions__control"
              disabled={true}
              icon="ecos-btn__i_right icon-small-down"
              id="group-actions-1-rec"
              invert={true}
              loaderType="points"
              loading={false}
            >
              
            </IcoBtn>,
            "contentComponent": "journal.group-actions.selected-records.label-count",
            "delay": 0,
            "hideArrow": false,
            "isOpen": false,
            "minWidthByContent": false,
            "placement": "top",
            "showAsNeeded": true,
            "target": "group-actions-1-rec",
            "text": "",
            "uncontrolled": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "",
              "className": "ecos-btn_hover_blue2 ecos-btn_grey3 ecos-group-actions__control",
              "disabled": true,
              "icon": "ecos-btn__i_right icon-small-down",
              "id": "group-actions-1-rec",
              "invert": true,
              "loaderType": "points",
              "loading": false,
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      false,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <DropdownOuter
            className="ecos-group-actions__dropdown"
            disabled={true}
            getStateOpen={[Function]}
            isStatic={true}
            itemClassName={[Function]}
            keyFields={
              Array [
                "id",
                "formRef",
                "pluralName",
                "_typeAct",
              ]
            }
            menuClassName="ecos-group-actions__dropdown-menu"
            modifiers={
              Object {
                "flip": Object {
                  "behavior": Array [
                    "bottom",
                    "top",
                    "right",
                    "left",
                  ],
                },
              }
            }
            onChange={[Function]}
            outClassName="ecos-dropdown-outer-tooltip"
            placement="bottom-start"
            source={Array []}
            titleField="pluralName"
            trigger="click"
            valueField="id"
          >
            <Tooltip
              autohide={false}
              boundariesElement="window"
              contentComponent="journal.group-actions.selected-records.label-count"
              delay={0}
              hideArrow={false}
              isOpen={false}
              minWidthByContent={false}
              placement="top"
              showAsNeeded={true}
              target="group-actions-1-rec"
              text=""
              uncontrolled={true}
            >
              <IcoBtn
                className="ecos-btn_hover_blue2 ecos-btn_grey3 ecos-group-actions__control"
                disabled={true}
                icon="ecos-btn__i_right icon-small-down"
                id="group-actions-1-rec"
                invert={true}
                loaderType="points"
                loading={false}
              >
                
              </IcoBtn>
            </Tooltip>
          </DropdownOuter>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Tooltip
              autohide={false}
              boundariesElement="window"
              contentComponent="journal.group-actions.selected-records.label-count"
              delay={0}
              hideArrow={false}
              isOpen={false}
              minWidthByContent={false}
              placement="top"
              showAsNeeded={true}
              target="group-actions-1-rec"
              text=""
              uncontrolled={true}
            >
              <IcoBtn
                className="ecos-btn_hover_blue2 ecos-btn_grey3 ecos-group-actions__control"
                disabled={true}
                icon="ecos-btn__i_right icon-small-down"
                id="group-actions-1-rec"
                invert={true}
                loaderType="points"
                loading={false}
              >
                
              </IcoBtn>
            </Tooltip>,
            "className": "ecos-group-actions__dropdown",
            "disabled": true,
            "getStateOpen": [Function],
            "isStatic": true,
            "itemClassName": [Function],
            "keyFields": Array [
              "id",
              "formRef",
              "pluralName",
              "_typeAct",
            ],
            "menuClassName": "ecos-group-actions__dropdown-menu",
            "modifiers": Object {
              "flip": Object {
                "behavior": Array [
                  "bottom",
                  "top",
                  "right",
                  "left",
                ],
              },
            },
            "onChange": [Function],
            "outClassName": "ecos-dropdown-outer-tooltip",
            "placement": "bottom-start",
            "source": Array [],
            "titleField": "pluralName",
            "trigger": "click",
            "valueField": "id",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autohide": false,
              "boundariesElement": "window",
              "children": <IcoBtn
                className="ecos-btn_hover_blue2 ecos-btn_grey3 ecos-group-actions__control"
                disabled={true}
                icon="ecos-btn__i_right icon-small-down"
                id="group-actions-1-rec"
                invert={true}
                loaderType="points"
                loading={false}
              >
                
              </IcoBtn>,
              "contentComponent": "journal.group-actions.selected-records.label-count",
              "delay": 0,
              "hideArrow": false,
              "isOpen": false,
              "minWidthByContent": false,
              "placement": "top",
              "showAsNeeded": true,
              "target": "group-actions-1-rec",
              "text": "",
              "uncontrolled": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "",
                "className": "ecos-btn_hover_blue2 ecos-btn_grey3 ecos-group-actions__control",
                "disabled": true,
                "icon": "ecos-btn__i_right icon-small-down",
                "id": "group-actions-1-rec",
                "invert": true,
                "loaderType": "points",
                "loading": false,
              },
              "ref": null,
              "rendered": "",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        false,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
